### PR TITLE
bug in truss element

### PR DESCRIPTION
**Pull request by Vahid-Galavi**
**📝 Description**
Solved a problem that leads to an internal error in the Intel compiler.

### CODE REVIEW

File: applications/GeoMechanicsApplication/custom_elements/geo_truss_element_base.cpp [hunk 1]
```diff
@@ -206,34 +207,36 @@ void GeoTrussElementBase<TDim,TNumNodes>::
...
 template< unsigned int TDim, unsigned int TNumNodes >
-void GeoTrussElementBase<TDim,TNumNodes>::
+ void GeoTrussElementBase<TDim,TNumNodes>::
     CalculateConsistentMassMatrix(MatrixType& rMassMatrix,
                                   const ProcessInfo& rCurrentProcessInfo) const
 {
-    KRATOS_TRY;
+    KRATOS_TRY
+
+    static constexpr unsigned int NDof = TDim * TNumNodes;
```
Comment (rubenzorrilla) [inline applications/GeoMechanicsApplication/custom_elements/geo_truss_element_base.cpp]: Isn't this unused?

File: applications/GeoMechanicsApplication/custom_elements/geo_truss_element_base.cpp [hunk 2]
```diff
@@ -206,34 +207,36 @@ void GeoTrussElementBase<TDim,TNumNodes>::
...
+
     const double A = GetProperties()[CROSS_AREA];
     const double L = GeoStructuralMechanicsElementUtilities::CalculateReferenceLength3D2N(*this);
     const double rho = GetProperties()[DENSITY];
 
-    const unsigned int factor = double(TDim * TNumNodes);
+    const double factor = double(TDim * TNumNodes);
```
Comment (rubenzorrilla) [inline applications/GeoMechanicsApplication/custom_elements/geo_truss_element_base.cpp]:
```suggestion
    const double factor = TDim * TNumNodes;
```
I think the cast is automatically done in this case.

File: applications/GeoMechanicsApplication/custom_elements/geo_truss_element_base.cpp (hunk 1 above)
Comment (Vahid-Galavi) [reply applications/GeoMechanicsApplication/custom_elements/geo_truss_element_base.cpp]: It is used in the same function.

Comment (rubenzorrilla) [reply applications/GeoMechanicsApplication/custom_elements/geo_truss_element_base.cpp]: For the future we can define these in the header, so we can use them all throughout the class.

Comment (Vahid-Galavi) [reply applications/GeoMechanicsApplication/custom_elements/geo_truss_element_base.cpp]: This variable is only used in this function. That's why I defined it in the function but I can also move it to the header file.

Comment (rubenzorrilla) [reply applications/GeoMechanicsApplication/custom_elements/geo_truss_element_base.cpp]: My point is that there is plenty of places in which the `TDim*TNumNodes` is used. In any case I'd leave this for the future (out of the scope of this PR) :wink:

Comment (Vahid-Galavi) [reply applications/GeoMechanicsApplication/custom_elements/geo_truss_element_base.cpp]: Anyway, I moved all static constexpr to the header file.

**Review by rubenzorrilla — APPROVED** [review]

**Comment by Vahid-Galavi** [comment]
Thanks, @rubenzorrilla for reviewing!

**Pull request merged**